### PR TITLE
Make scion code l2-address-agnostic, introduce SVC

### PR DESCRIPTION
**Pull request by kormat**
- Add lib.packet.host_addr to encapsulate all host addresses.
- Update lib.dnsclient to return host addresses.
- Add new "SVC" pseudo-address space, for sending packets to SCION /services/.
- Update the scion common header to store address /types/ rather than lengths.
- Update lib.topology to present addresses as host address.

More work needed:
- Add network library, so that packet sending/receiving can be address-agnostic.
- DNS client needs to support AAAA records.
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/327%23issuecomment-132188068%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23issuecomment-132235819%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23issuecomment-132498828%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23issuecomment-132649348%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37392094%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37431517%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37513164%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37513621%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37513888%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37513918%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37521116%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37521211%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37521234%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37522400%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37522453%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37522454%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37526090%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37530157%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23issuecomment-133012359%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23issuecomment-133013132%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23issuecomment-132188068%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22HostAddrType%20of%20management%20packets%20%28non-data%20packets%29%20is%20SVC%2C%20and%20HostAddrType%20of%20Data%20packet%20is%20IPv4%20or%20IPv6.%20Is%20it%20right%3F%5Cr%5CnCan%20the%20router%20treat%20the%20packet%20as%20Data%20packet%20without%20checking%20the%20%20src/dst%20addresses%20%20in%20case%20that%20HostAddrType%20is%20IPv4%20or%20IPv6%3F%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-08-18T12%3A08%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9196745%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/taka-sasaki%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3E%20HostAddrType%20of%20management%20packets%20%28non-data%20packets%29%20is%20SVC%2C%20and%20HostAddrType%20of%20Data%20packet%20is%20IPv4%20or%20IPv6.%20Is%20it%20right%3F%5Cr%5Cn%5Cr%5CnYep%2C%20correct.%5Cr%5Cn%5Cr%5Cn%3E%20Can%20the%20router%20treat%20the%20packet%20as%20Data%20packet%20without%20checking%20the%20%20src/dst%20addresses%20%20in%20case%20that%20HostAddrType%20is%20IPv4%20or%20IPv6%3F%5Cr%5Cn%5Cr%5CnI%20believe%20so%2C%20yes.%22%2C%20%22created_at%22%3A%20%222015-08-18T14%3A42%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22lgtm%22%2C%20%22created_at%22%3A%20%222015-08-19T08%3A56%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27ve%20substantially%20rewritten%20this%20since%20you%20last%20looked%2C%20so%20if%20you%20can%20give%20it%20another%20pass%2C%20that%20would%20be%20great.%22%2C%20%22created_at%22%3A%20%222015-08-19T15%3A58%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22lgtm%22%2C%20%22created_at%22%3A%20%222015-08-20T13%3A46%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%5C%5Co/%22%2C%20%22created_at%22%3A%20%222015-08-20T13%3A49%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20a67e58fd27f83302dfdf742e57abfe871b615854%20lib/dnsclient.py%2031%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37513164%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22maybe%20a%20comment%20that%20we%20should%20respect%20DNS%20TTL%20at%20some%20point%22%2C%20%22created_at%22%3A%20%222015-08-20T09%3A57%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222015-08-20T12%3A00%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-20T12%3A21%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/dnsclient.py%3AL12-60%22%7D%2C%20%22Pull%2088b8ee4690cc72b46cc1df518e20affe28317ca7%20infrastructure/scion_elem.py%2035%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37392094%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22does%20not%20look%20too%20agnostic%20%3B-%29%22%2C%20%22created_at%22%3A%20%222015-08-19T08%3A46%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20updated%20code%20adds%20a%20comment%20here%20explaining%20the%20problem.%22%2C%20%22created_at%22%3A%20%222015-08-19T15%3A54%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-08-20T10%3A07%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/scion_elem.py%3AL251-258%22%7D%2C%20%22Pull%20a67e58fd27f83302dfdf742e57abfe871b615854%20topology/generator.py%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37513888%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22not%20sure%20with%20this%20%60ToAddrType%60.%20This%20is%20ER2ER%20communication%2C%20and%20addressing%20here%20has%20to%20be%20consistent.%22%2C%20%22created_at%22%3A%20%222015-08-20T10%3A07%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ah%2C%20you%27re%20saying%20that%20the%20type%20of%20%60ToAddr%60%20is%20implicitly%20the%20same%20as%20for%20%60Addr%60%3F%22%2C%20%22created_at%22%3A%20%222015-08-20T12%3A01%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%5Cn%5CnOn%2020%20August%202015%20at%2014%3A01%2C%20Stephen%20Shirley%20%3Cnotifications%40github.com%3E%5Cnwrote%3A%5Cn%5Cn%3E%20In%20topology/generator.py%5Cn%3E%20%3Chttps%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37521234%3E%3A%5Cn%3E%5Cn%3E%20%3E%20%40%40%20-467%2C6%20%2B467%2C7%20%40%40%20def%20write_topo_files%28self%2C%20ad_configs%2C%20er_ip_addresses%29%3A%5Cn%3E%20%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%27NeighborType%27%3A%20nbr_type%2C%5Cn%3E%20%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%27AddrType%27%3A%20%27IPv4%27%2C%5Cn%3E%20%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%27Addr%27%3A%20ip_address_pub%2C%5Cn%3E%20%3E%20%2B%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%27ToAddrType%27%3A%20%27IPv4%27%2C%5Cn%3E%5Cn%3E%20Ah%2C%20you%27re%20saying%20that%20the%20type%20of%20ToAddr%20is%20implicitly%20the%20same%20as%20for%5Cn%3E%20Addr%3F%5Cn%3E%5Cn%3E%20%5Cu2014%5Cn%3E%20Reply%20to%20this%20email%20directly%20or%20view%20it%20on%20GitHub%5Cn%3E%20%3Chttps%3A//github.com/netsec-ethz/scion/pull/327/files%23r37521234%3E.%5Cn%3E%5Cn%22%2C%20%22created_at%22%3A%20%222015-08-20T12%3A20%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ok%20cool%2C%20fixed%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-08-20T13%3A05%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-08-20T13%3A46%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20topology/generator.py%3AL467-474%22%7D%2C%20%22Pull%20a67e58fd27f83302dfdf742e57abfe871b615854%20lib/topology.py%2029%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/327%23discussion_r37513621%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20not%20%60None%60%20%3F%22%2C%20%22created_at%22%3A%20%222015-08-20T10%3A03%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Any%20immutable%20object%20works%2C%20i%20just%20went%20with%20tuple%20here%20as%20it%20helps%20indicate%20the%20expected%20type.%22%2C%20%22created_at%22%3A%20%222015-08-20T12%3A01%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-20T12%3A21%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/topology.py%3AL35-55%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/327#issuecomment-132188068'>General Comment</a></b>
- <a href='https://github.com/taka-sasaki'></a> HostAddrType of management packets (non-data packets) is SVC, and HostAddrType of Data packet is IPv4 or IPv6. Is it right?
  Can the router treat the packet as Data packet without checking the  src/dst addresses  in case that HostAddrType is IPv4 or IPv6?
- <a href='https://github.com/kormat'></a> > HostAddrType of management packets (non-data packets) is SVC, and HostAddrType of Data packet is IPv4 or IPv6. Is it right?
  Yep, correct.
  > Can the router treat the packet as Data packet without checking the  src/dst addresses  in case that HostAddrType is IPv4 or IPv6?
  I believe so, yes.
- <a href='https://github.com/pszalach'></a> lgtm
- <a href='https://github.com/kormat'></a> I've substantially rewritten this since you last looked, so if you can give it another pass, that would be great.
- <a href='https://github.com/pszalach'></a> lgtm
- <a href='https://github.com/kormat'></a> \o/
- [x] <a href='#crh-comment-Pull 88b8ee4690cc72b46cc1df518e20affe28317ca7 infrastructure/scion_elem.py 35'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/327#discussion_r37392094'>File: infrastructure/scion_elem.py:L251-258</a></b>
- <a href='https://github.com/pszalach'></a> does not look too agnostic ;-)
- <a href='https://github.com/kormat'></a> The updated code adds a comment here explaining the problem.
- [x] <a href='#crh-comment-Pull a67e58fd27f83302dfdf742e57abfe871b615854 lib/dnsclient.py 31'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/327#discussion_r37513164'>File: lib/dnsclient.py:L12-60</a></b>
- <a href='https://github.com/pszalach'></a> maybe a comment that we should respect DNS TTL at some point
- <a href='https://github.com/kormat'></a> Done
- [x] <a href='#crh-comment-Pull a67e58fd27f83302dfdf742e57abfe871b615854 lib/topology.py 29'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/327#discussion_r37513621'>File: lib/topology.py:L35-55</a></b>
- <a href='https://github.com/pszalach'></a> why not `None` ?
- <a href='https://github.com/kormat'></a> Any immutable object works, i just went with tuple here as it helps indicate the expected type.
- [x] <a href='#crh-comment-Pull a67e58fd27f83302dfdf742e57abfe871b615854 topology/generator.py 4'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/327#discussion_r37513888'>File: topology/generator.py:L467-474</a></b>
- <a href='https://github.com/pszalach'></a> not sure with this `ToAddrType`. This is ER2ER communication, and addressing here has to be consistent.
- <a href='https://github.com/kormat'></a> Ah, you're saying that the type of `ToAddr` is implicitly the same as for `Addr`?
- <a href='https://github.com/pszalach'></a> Yes
  On 20 August 2015 at 14:01, Stephen Shirley notifications@github.com
  wrote:
  > In topology/generator.py
  > https://github.com/netsec-ethz/scion/pull/327#discussion_r37521234:
  >
  > > @@ -467,6 +467,7 @@ def write_topo_files(self, ad_configs, er_ip_addresses):
  > >                                    'NeighborType': nbr_type,
  > >                                    'AddrType': 'IPv4',
  > >                                    'Addr': ip_address_pub,
  > > +                                  'ToAddrType': 'IPv4',
  >
  > Ah, you're saying that the type of ToAddr is implicitly the same as for
  > Addr?
  >
  > —
  > Reply to this email directly or view it on GitHub
  > https://github.com/netsec-ethz/scion/pull/327/files#r37521234.
  >
- <a href='https://github.com/kormat'></a> Ok cool, fixed :)

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/327?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/327?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/327'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: infrastructure/scion_elem.py [hunk 1]
```diff
@@ -256,7 +251,7 @@ def dns_query(self, qname, default):
         valid instead.
         """
         try:
-            return self.dns.query(qname)[0]
+            return parse_host_addr("IPv4", self.dns.query(qname)[0])
```
Comment (pszal) [inline infrastructure/scion_elem.py]: does not look too agnostic ;-)

Comment (kormat) [inline infrastructure/scion_elem.py]: The updated code adds a comment here explaining the problem.

File: lib/dnsclient.py [hunk 2]
```diff
@@ -12,47 +12,48 @@
...
 # SCION
 from lib.defines import SCION_DNS_PORT
 from lib.errors import SCIONBaseError
+from lib.packet.host_addr import haddr_parse
 
-#: Number of records to cache
+#: Number of records to cache.
 DNS_CACHE_MAX_SIZE = 100
-#: Seconds
+#: Cache validity in seconds.
```
Comment (pszal) [inline lib/dnsclient.py]: maybe a comment that we should respect DNS TTL at some point

File: lib/topology.py [hunk 3]
```diff
@@ -35,34 +35,20 @@ class Element(object):
...
-    :ivar addr: IP or SCION address of a server or edge router.
-    :type addr: :class:`IPv4Address` or :class:`IPv6Address`
-    :ivar name: element name or id
-    :type name: str
+    :ivar HostAddrBase addr: Host address of a server or edge router.
+    :ivar str name: element name or id
     """
 
-    def __init__(self, addr=None, name=None):
+    def __init__(self, addr_info=(), name=None):
```
Comment (pszal) [inline lib/topology.py:42]: why not `None` ?

File: topology/generator.py [hunk 4]
```diff
@@ -467,6 +467,7 @@ def write_topo_files(self, ad_configs, er_ip_addresses):
                                   'NeighborType': nbr_type,
                                   'AddrType': 'IPv4',
                                   'Addr': ip_address_pub,
+                                  'ToAddrType': 'IPv4',
```
Comment (pszal) [inline topology/generator.py]: not sure with this `ToAddrType`. This is ER2ER communication, and addressing here has to be consistent.

File: infrastructure/scion_elem.py (hunk 1 above)
Comment (pszal) [inline infrastructure/scion_elem.py]: :+1:

File: lib/dnsclient.py (hunk 2 above)
Comment (kormat) [inline lib/dnsclient.py]: Done

File: lib/topology.py (hunk 3 above)
Comment (kormat) [inline lib/topology.py:42]: Any immutable object works, i just went with tuple here as it helps indicate the expected type.

File: topology/generator.py (hunk 4 above)
Comment (kormat) [inline topology/generator.py]: Ah, you're saying that the type of `ToAddr` is implicitly the same as for `Addr`?

Comment (pszal) [inline topology/generator.py]: Yes

File: lib/dnsclient.py (hunk 2 above)
Comment (pszal) [inline lib/dnsclient.py]: :+1:<a href='#crh-update-none'></a>

File: lib/topology.py (hunk 3 above)
Comment (pszal) [inline lib/topology.py:42]: :+1:<a href='#crh-update-complete'></a>

File: topology/generator.py (hunk 4 above)
Comment (kormat) [inline topology/generator.py]: Ok cool, fixed :)

Comment (pszal) [inline topology/generator.py]: :+1:

**Pull request merged**